internal/file/tar_file_traversal.go

```diff
@@ -0,0 +1,77 @@
+package file
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"path/filepath"
+
+	"github.com/bmatcuk/doublestar/v4"
+	"github.com/mholt/archiver/v3"
+)
+
+// ExtractGlobsFromTarToUniqueTempFile extracts paths matching the given globs within the given archive to a temporary directory, returning file openers for each file extracted.
+func ExtractGlobsFromTarToUniqueTempFile(archivePath, dir string, globs ...string) (map[string]Opener, error) {
+	results := make(map[string]Opener)
+
+	// don't allow for full traversal, only select traversal from given paths
+	if len(globs) == 0 {
+		return results, nil
+	}
+
+	visitor := func(file archiver.File) error {
+		defer file.Close()
+
+		// ignore directories
+		if file.FileInfo.IsDir() {
+			return nil
+		}
+
+		// ignore any filename that doesn't match the given globs...
+		if !matchesAnyGlob(file.Name(), globs...) {
+			return nil
+		}
+
+		// we have a file we want to extract....
+		tempfilePrefix := filepath.Base(filepath.Clean(file.Name())) + "-"
+		tempFile, err := ioutil.TempFile(dir, tempfilePrefix)
```

Comment:
Not silly at all --this has been optimized for memory consumption with the assumption that disk storage is available and cheap... but there may be room for improvement here.

Since all jars are zip files, we can seek to the end of the file to get the table of contents and search against the list of file names within the TOC to see if there are any files we should parse. Then we decompress and extract the specific files of interest to be parsed. 

The biggest trick is that the java archive parser accepts a `io.Reader`, which once read we can't go back. Since the TOC is at the end we'd need to re-open the file to get the contents. Since the parser doesn't have access to the file, it is easier to copy out the jar to a temp file so that we can extract specific files and parse them there. 

There are probably other approaches that can be taken, but so far haven't been obvious [to me] and are out of scope for this PR (but another draft PR to try out different approaches here is welcome!)